include/vcpkg/base/json.h

```diff
@@ -209,11 +210,11 @@ namespace vcpkg::Json
         ~Object() = default;
 
         // asserts if the key is found
-        template<class StringViewLike,
-                 class = std::enable_if_t<std::is_constructible<StringView, StringViewLike&&>::value>>
-        Value& insert(StringView key, StringViewLike&& value)
+        template<class StringLike,
+                 std::enable_if_t<std::is_constructible<StringView, const StringLike&>::value, int> = 0>
+        Value& insert(StringView key, const StringLike& value)
         {
-            return this->insert(key, StringView(static_cast<StringViewLike&&>(value)).to_string());
+            return this->insert(key, StringView(value).to_string());
```

Comment:
I believe this should still have a `static_cast<const StringLike&>` to capture string literal lengths.

Reply:
That cast has no effect: https://gcc.godbolt.org/z/nfc6rqo3W

Reply:
TIL, and am less happy as a result